core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1825,6 +1825,43 @@ public void testConstructV3Metadata() {
         3);
   }
 
+  @Test
+  public void testV3GeometryTypeSupport() {
```

Comment:
I think a better place for this to be tested in in `TestSchema.testUnsupportedTypes`. There are already a good set of cases for nesting and other types.

Reply:
We've already updated `TestSchema` to make use of `testUnsupportedTypes` and `testTypeSupported` to test geospatial types. I'm removing this test case from `TestTableMetadata`.